src/main/java/io/confluent/connect/jdbc/sink/metadata/FieldsMetadata.java

```diff
@@ -56,6 +56,10 @@ private FieldsMetadata(
     this.allFields = allFields;
   }
 
+  public Map<String, SinkRecordField> getAllFields() {
+    return allFields;
+  }
+
```

Comment:
We are able to access the `FieldsMetadata` object in our tests, so I already do mock the object. The problem is that `allFields` was originally accessed through calling it directly as a field of the object, which means I can't give an expected return object when it is called. In other words, I want to mock out what `FieldsMetadata.allFields` returns but this can't be done since `allFields` is a field. So rather, I made it into a method so I can mock out the return.

Reply:
Sorry, I'm a little confused--is it strictly necessary to mock this at all or can a bona fide `FieldsMetadata` object be constructed instead? It seems like the constructor logic is fairly lightweight and if all we're looking for is to create a `FieldsMetadata` instance whose `allFields` field has a certain value, what value is added by mocking? It seems like a cleaner change to upgrade the visibility of the constructor instead, but maybe I'm missing something.

Reply:
Basically, all the interaction that is done in the tested method with `FieldsMetadata` can be ignored and it is easier to test if this is the case, so I just want to stub all of its functionality. I'm not quite sure I understand your suggestion, so let me know if what you are saying helps with this case.